### PR TITLE
Revamp PR template

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Introduced a new PR template format to improve clarity and provide more prompts/guidance through comments.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
This PR uses the new template!

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Would appreciate feedback on:
- clarity of wording
- if I should add details to any of the comments
- suggestions on sections to add/remove/modify

Feel free to modify on this branch directly!

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
n/a

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
n/a

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
Mirrors changes from https://github.com/calblueprint/vouchers4veggies/pull/39

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
CC: @davidqing6432

### CODE REVIEW

**Review by david-qing — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
+[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
+
+## Relevant Links :link:
 
+### Online sources :information_source:
 
+[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
 
+### Related PRs :raised_hands:
+
```
Comment (wangannie) [inline .github/pull_request_template.md:28]: I would remove the newline here (so the comment directly follows the header)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
+[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
+[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
```
Comment (wangannie) [inline .github/pull_request_template.md:30]: I would add a couple of lines between the "Optional" and the "This tags the PL" comment so it's clear they're supposed to type between the lines
